pkg/cache/cache.go

```diff
@@ -142,6 +151,11 @@ func (c *serviceAccountCache) ToJSON() string {
 
 func (c *serviceAccountCache) addSA(sa *v1.ServiceAccount) {
 	arn, ok := sa.Annotations[c.annotationPrefix+"/"+pkg.RoleARNAnnotation]
+
+	if !strings.Contains(arn, "arn:") && c.composeRoleArn.Enabled {
+		arn = fmt.Sprintf("arn:%s:iam::%s:role/%s", c.composeRoleArn.Partition, c.composeRoleArn.AccountID, arn)
+	}
+
```

Comment:
I guess we don't validate the `role-arn` annotation that I see.  After this if block, it would be nice if we compared the value of `arn` to an ARN regex and logged an error if it doesn't match the regex.  Just in case someone misinterprets this new parameter, or has a typo, and also just because we should have had validation in the first place.  (Unless I'm missing it, I didn't look closely).  

Here is some code and a possible regex you can use:

```suggestion
	if !strings.Contains(arn, "arn:") && c.composeRoleArn.Enabled {
		arn = fmt.Sprintf("arn:%s:iam::%s:role/%s", c.composeRoleArn.Partition, c.composeRoleArn.AccountID, arn)
	}

	matched, err := regexp.Match(`^arn:aws[a-z0-9-]*:iam::\d{12}:role\/[\w-\/.@+=,]+$`, []byte(arn))
	if err != nil {
		klog.Errorf("Regex error: %v", err)
	} else if !matched {
		klog.Warningf("arn is invalid: %s", arn)
	}
```

What do you think?